### PR TITLE
chore: add upgrade testing and task cleanups

### LINKED ISSUE #58: Implement simple upgrade test for the gitlab repo and add to the job matrix to facilitate running this in CI
### Is your feature request related to a problem? Please describe.
We need to test basic upgrade processes of the single packages for SWF in order to increase confidence and hedge against risk of burning compute and cycles when testing upgrades in the SWF integration repo. This is currently lacking in the single package repos and is necessary for us to be efficient in our work. This issue is to create a set of uds tasks that will run an upgrade process and then update CI to run install and upgrade tests in this repo. For an example look at how it is being done in uds core for the full tests in that repo:

https://github.com/defenseunicorns/uds-core/blob/d2d005b9caf0b067f850f752a3d6e643c99a74c6/tasks/test.yaml#L36

also for reference the matrix of that (just a reference, for us it should be less complicated for these package repos as they are not mono repos) https://github.com/defenseunicorns/uds-core/blob/d2d005b9caf0b067f850f752a3d6e643c99a74c6/.github/workflows/pull-request-conditionals.yaml#L64

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This adds upgrade-tests to the GitLab package. (blocked on https://github.com/defenseunicorns/uds-common/pull/70)

## Related Issue

Fixes #58 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-package-gitlab/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: tasks/publish.yaml [hunk 1]
```diff
@@ -18,11 +19,15 @@ tasks:
   - name: package
     description: Build and publish the packages
     actions:
-      - description: Create the packages
-        cmd: |
-          set -e
-          ZARF_ARCHITECTURE=amd64 uds run -f tasks/create.yaml gitlab-package --set FLAVOR=${FLAVOR}
-          ZARF_ARCHITECTURE=arm64 uds run -f tasks/create.yaml gitlab-package --set FLAVOR=${FLAVOR}
+      - description: Create the AMD64 package
```
Comment (zachariahmiller) [inline tasks/publish.yaml:22]: yeah wtb matrix 😢  but looks good for now!

**Review by zachariahmiller — COMMENTED** [review]
Are we looking to merge the common updates before this and then bump the included tasks versions? this looks good pending that and changes to the actions to add an upgrade test. just lmk when its ready

**Comment by Racer159** [comment]
> Are we looking to merge the common updates before this and then bump the included tasks versions? this looks good pending that and changes to the actions to add an upgrade test. just lmk when its ready

Yes, still going through stuff in btwn mtgs

**Comment by Racer159** [comment]
Note I changed the names for how the marix works - welcome feedback on what those should be across our repos

**Comment by zachariahmiller** [comment]
@Racer159 uds-common 0.3.1 is released **https://github.com/defenseunicorns/uds-common/releases/tag/v0.3.1**

**Review by zachariahmiller — APPROVED** [review]
Required Checks need to get updated again but looks good as far as i can tell. :shipit:

**Pull request merged**